sdk/core/Azure.Core/samples/DynamicContent.md

```diff
@@ -0,0 +1,73 @@
+# Azure SDK Dynamic JSON samples
+
+Azure SDK client [Protocol methods](ProtocolMethods.md) do not take or return model types.  JSON response content can be accessed using BCL JSON types, but use of these APIs can result in code that is difficult to read and obscures the author's intent.  To improve the developer experience, Azure.Core provides a dynamic layer over JSON APIs.
```

Comment:
Yes, I like what you're doing there.  Let me take another iteration over that given your suggestions.

Also, good call out on links to .NET dynamic docs!  In the past, I've gotten feedback that I link out too much so I was trying to work against that instinct here, but I think this is a good place to do so for a basic and fundamental concept that may be unfamiliar.  Thanks!

Reply:
Let's just make sure that the dynamic docs don't talk too much about how dynamic internals work and how IDynamicMetaObject is implemented. We don't need people to understand it.

Reply:
Might be worth mentioning at least that we implement it, so that devs know what they can/can't do. Or perhaps that's implied enough that they can use `dynamic`?